[BE] reduce log spew from test_triton_kernels.py

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #145895

One of the tests in this file was setting `self._logging.set_logs(output_code=True)` - which would cause logs to be printed for the rest of the tests in this file.

This PR puts the log-setting in a context manager so that the old behavior is restored afterwards.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

davidberard98 committed:
[BE] reduce log spew from test_triton_kernels.py

One of the tests in this file was setting `self._logging.set_logs(output_code=True)` - which would cause logs to be printed for the rest of the tests in this file.

This PR puts the log-setting in a context manager so that the old behavior is restored afterwards.

[ghstack-poisoned]

Skylion007 commented on test/inductor/test_triton_kernels.py:
```diff
@@ -3504,9 +3505,18 @@ def grid(META):
+            finally:
+                torch._logging.set_logs(output_code=False)
+
+        with output_code_logs():
+            with self.assertLogs(logger="torch._inductor", level=logging.DEBUG) as log:
+                foo(x, w)
```
```suggestion
        with output_code_logs(), self.assertLogs(logger="torch._inductor", level=logging.DEBUG) as log:
            foo(x, w)

Skylion007 commented on test/inductor/test_triton_kernels.py:
```diff
@@ -3504,9 +3505,18 @@ def grid(META):
-        with self.assertLogs(logger="torch._inductor", level=logging.DEBUG) as log:
-            foo(x, w)
+
+        @contextlib.contextmanager
```
Nit should be an enable output_code_logs() context manager that restore the original state.

davidberard98 replied:
thanks for the review @Skylion007 !

quick question about this nit - did you mean that this context manager should restore the original state instead of restoring to `output_logs=False` ?

Skylion007 replied:
yep, just in case. good code design.

davidberard98 replied:
I agree this would be ideal, but I don't see an easy way to obtain the original state without making some more changes in `torch._logging`

davidberard98 replied:
thanks @SamGinzburg for the suggestion to use `logging_utils`, which has a context manager for patching and restoring logging settings

davidberard98 committed:
Update on "[BE] reduce log spew from test_triton_kernels.py"


One of the tests in this file was setting `self._logging.set_logs(output_code=True)` - which would cause logs to be printed for the rest of the tests in this file.

This PR puts the log-setting in a context manager so that the old behavior is restored afterwards.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

davidberard98 committed:
Update on "[BE] reduce log spew from test_triton_kernels.py"


One of the tests in this file was setting `self._logging.set_logs(output_code=True)` - which would cause logs to be printed for the rest of the tests in this file.

This PR puts the log-setting in a context manager so that the old behavior is restored afterwards.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

nmacchioni reviewed (approved):
LGTM!

davidberard98 commented:
@pytorchbot merge
